Apps/Common/SIMMultiPatchModelGen.h

```diff
@@ -0,0 +1,49 @@
+// $Id$
+//==============================================================================
+//!
+//! \file SIMMultiPatchModelGen.h
+//!
+//! \date Sep 5 2016
+//!
+//! \author Arne Morten Kvarving / SINTEF
+//!
+//! \brief Base class for simulators equipped with multi-patch model generators.
+//!
+//==============================================================================
+
+#ifndef _SIM_MULTPATCH_MODEL_GEN_H_
+#define _SIM_MULTIPATCH_MODEL_GEN_H_
+
+#include "SIMbase.h"
+
+class TiXmlElement;
+
+
+//! \brief Inherit this class to equip your SIM multi-patch model generators.
+template<class Dim>
+class SIMMultiPatchModelGen : public Dim {
+public:
+  //! \brief Default constructor.
+  //! \param[in] n1 Number of fields
+  //! \param[in] check If \e true, ensure the model is in a right-hand system
+  SIMMultiPatchModelGen(int n1,
+                        bool checkRHS=false) :
```

Comment:
checkRHS er overflødig her, da den kun brukes av readPatch.. metodene som vel ikke brukes når modellen blir generert. Kan derfor her skrive:

SIMMultiPatchModelGen(int n1) : Dim(n1) {}

og tilsvarende for mixed-constructoren under.